From 51abcbf5c17c860da5092a485cb8bd4b37a6d51b Mon Sep 17 00:00:00 2001
From: Frank Wang <frank.wang@rock-chips.com>
Date: Wed, 5 Apr 2017 09:35:37 +0800
Subject: [PATCH] soc: rockchip: amend rk3368-mbox related *.h to soc/rockchip

This adds move rk3368-mbox related *.h files from linux/rockchip
to soc/rockchip.

Change-Id: I2d57b6baa64d531e89766a5384d8c217cf347ebf
Signed-off-by: Frank Wang <frank.wang@rock-chips.com>
---
 drivers/mailbox/rk3368-mailbox.c        |  6 +++---
 drivers/mailbox/scpi_protocol.c         |  6 +++---
 include/linux/rockchip/rk3368-mailbox.h | 21 ---------------------
 include/soc/rockchip/rk3368-mailbox.h   | 22 ++++++++++++++++++++++
 include/{linux => soc}/rockchip/scpi.h  | 14 ++++++--------
 5 files changed, 34 insertions(+), 35 deletions(-)
 delete mode 100644 include/linux/rockchip/rk3368-mailbox.h
 create mode 100644 include/soc/rockchip/rk3368-mailbox.h
 rename include/{linux => soc}/rockchip/scpi.h (91%)

diff --git a/drivers/mailbox/rk3368-mailbox.c b/drivers/mailbox/rk3368-mailbox.c
index 34263d7453d2..03f3a74d4623 100644
--- a/drivers/mailbox/rk3368-mailbox.c
+++ b/drivers/mailbox/rk3368-mailbox.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2015, Fuzhou Rockchip Electronics Co., Ltd
+ * Copyright (c) 2017, Fuzhou Rockchip Electronics Co., Ltd
  *
  * This program is free software; you can redistribute it and/or modify it
  * under the terms and conditions of the GNU General Public License,
@@ -22,8 +22,8 @@
 #include <linux/slab.h>
 #include <linux/clk.h>
 
-#include <linux/rockchip/rk3368-mailbox.h>
-#include <linux/rockchip/scpi.h>
+#include <soc/rockchip/rk3368-mailbox.h>
+#include <soc/rockchip/scpi.h>
 
 #define MAILBOX_VERSION			"V1.00"
 
diff --git a/drivers/mailbox/scpi_protocol.c b/drivers/mailbox/scpi_protocol.c
index 9cc5384e070e..4ace0a2d4f62 100644
--- a/drivers/mailbox/scpi_protocol.c
+++ b/drivers/mailbox/scpi_protocol.c
@@ -2,7 +2,7 @@
  * System Control and Power Interface (SCPI) Message Protocol driver
  *
  * Copyright (C) 2014 ARM Ltd.
- * Copyright (c) 2014, Fuzhou Rockchip Electronics Co., Ltd
+ * Copyright (c) 2017, Fuzhou Rockchip Electronics Co., Ltd
  *
  * This program is free software; you can redistribute it and/or modify it
  * under the terms and conditions of the GNU General Public License,
@@ -25,8 +25,8 @@
 #include <linux/printk.h>
 #include <linux/mailbox_client.h>
 #include <linux/slab.h>
-#include <linux/rockchip/scpi.h>
-#include <linux/rockchip/rk3368-mailbox.h>
+#include <soc/rockchip/scpi.h>
+#include <soc/rockchip/rk3368-mailbox.h>
 
 #define SCPI_VERSION		0x01000002	/* version: 1.0.0.2 */
 
diff --git a/include/linux/rockchip/rk3368-mailbox.h b/include/linux/rockchip/rk3368-mailbox.h
deleted file mode 100644
index 659375c7d874..000000000000
--- a/include/linux/rockchip/rk3368-mailbox.h
+++ /dev/null
@@ -1,21 +0,0 @@
-/*
- * Copyright (c) 2014, Fuzhou Rockchip Electronics Co., Ltd
- *
- * This program is free software; you can redistribute it and/or modify it
- * under the terms and conditions of the GNU General Public License,
- * version 2, as published by the Free Software Foundation.
- */
-
-#ifndef __MAILBOX_ROCKCHIP_H__
-#define __MAILBOX_ROCKCHIP_H__
-
-struct rk3368_mbox_msg {
-	u32 cmd;
-	int tx_size;
-	void *tx_buf;
-	int rx_size;
-	void *rx_buf;
-	void *cl_data;
-};
-
-#endif /* __MAILBOX_ROCKCHIP_H__ */
diff --git a/include/soc/rockchip/rk3368-mailbox.h b/include/soc/rockchip/rk3368-mailbox.h
new file mode 100644
index 000000000000..77208277f419
--- /dev/null
+++ b/include/soc/rockchip/rk3368-mailbox.h
@@ -0,0 +1,22 @@
+/*
+ * Copyright (C) 2017, Fuzhou Rockchip Electronics Co., Ltd
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+#ifndef __RK3368_MAILBOX_H__
+#define __RK3368_MAILBOX_H__
+
+struct rk3368_mbox_msg {
+	u32 cmd;
+	int tx_size;
+	void *tx_buf;
+	int rx_size;
+	void *rx_buf;
+	void *cl_data;
+};
+
+#endif /* __RK3368_MAILBOX_H__ */
diff --git a/include/linux/rockchip/scpi.h b/include/soc/rockchip/scpi.h
similarity index 91%
rename from include/linux/rockchip/scpi.h
rename to include/soc/rockchip/scpi.h
index 468a35095c13..a2fc901f9cd2 100644
--- a/include/linux/rockchip/scpi.h
+++ b/include/soc/rockchip/scpi.h
@@ -1,8 +1,5 @@
 /*
- * SCPI Message Protocol driver header
- *
- * Copyright (C) 2014 ARM Ltd.
- * Copyright (C) 2014, Fuzhou Rockchip Electronics Co., Ltd
+ * Copyright (C) 2017, Fuzhou Rockchip Electronics Co., Ltd
  *
  * This program is free software; you can redistribute it and/or modify it
  * under the terms and conditions of the GNU General Public License,
@@ -12,11 +9,11 @@
  * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
  * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
  * more details.
- *
- * You should have received a copy of the GNU General Public License along with
- * this program. If not, see <http://www.gnu.org/licenses/>.
  */
-#include <linux/types.h>
+
+#ifndef __SOC_ROCKCHIP_SCPI_H
+#define __SOC_ROCKCHIP_SCPI_H
+
 #include <linux/rockchip/common.h>
 
 #ifdef CONFIG_RK3368_SCPI_PROTOCOL
@@ -151,3 +148,4 @@ static inline int scpi_thermal_set_clk_cycle(u32 cycle)
 	return -EPERM;
 }
 #endif
+#endif
-- 
2.35.3

